packages/stream-server-uws/stream_server_uws.js

```diff
@@ -0,0 +1,90 @@
+// noinspection NpmUsedModulesInstalled
+const { RoutePolicy } = require('meteor/routepolicy');
+// noinspection NpmUsedModulesInstalled
+const { WebApp } = require('meteor/webapp');
+
+// Try load of `uws` Node.js package
+/** @type {WebSocket} */
+let WebSocket;
+try {
+  // noinspection NpmUsedModulesInstalled
+  WebSocket = require('sc-uws');
+} catch (e) {
+  console.warn('Run:\nmeteor npm install sc-uws')
```

Comment:
You should check that the error here is actually due to `sc-uws` not being installed, rather than some error thrown while trying to evaluate the package.

One way to do this is to use `require.resolve` first, which returns the absolute module identifier if the package is installed, without actually importing it, and otherwise throws an error:

```js
let uwsAbsId;
try {
  uwsAbsId = require.resolve("sc-uws");
} catch (e) {
  console.warn(...);
}
if (typeof uwsAbsId === "string") {
  WebSocket = require(uwsAbsId);
}
```

Reply:
Okay, I will change that.

Reply:
@benjamn I found difference between how works the `require.resolve()` in Node.js and Meteor.

So, your code not working if module is not installed.

## Code

```js
let path;
try {
  path = require.resolve('uws');
} catch (e) {
  console.error(e);
}
console.log('path', path);
```

## Output if installed

```
path /node_modules/uws/uws.js
```

## Output if not installed

### Node.js

```
{ [Error: Cannot find module 'uws'] code: 'MODULE_NOT_FOUND', name: 'Error' }
path undefined
```

### Meteor

```
path uws
```

## Explanation

In Meteor `require.resolve()` was patched in the file `modules-runtime.js`:

```js
  function makeRequire(file) {
    var module = file.module;

    function require(id) {
      return module.require(id);
    }

    require.extensions = fileGetExtensions(file).slice(0);

    require.resolve = function resolve(id) {
      return module.resolve(id);
    };

    return require;
  }
```

We get patched version of `require()` if using packages `modules` or `ecmascript`.